### PR TITLE
Allow users to run CrashMonkey and workload as separate processes.

### PR DESCRIPTION
**Pull request by ashmrtn**
New functionality can be used by supplying the `-b` flag to CrashMonkey when starting. Then, in another shell, run the setup and workload you want. The programs in `build/user_tools` allow you to tell CrashMonkey what to do next. The general order for running them when running CrashMonkey with `-b` is:
* run pre-test setup (user defined)
* run `build/user_tools/begin_log`
* run user workload
* run `build/user_tools/end_log`
* run `build/user_tools/begin_tests`

Example of how to run, with the same "user workload" as a smoke test that already exists:
* shell 1: `cd crashmonkey/code`
* shell 1: `make default tests/echo_sub_dir.so`
* shell 1: `sudo ../build/harness -f /dev/sda -t ext4 -m barrier -d /dev/cow_ram0 -e 10240 -b tests/echo_sub_dir.so`
* shell 2: `sudo mkdir /mnt/snapshot/test_dir`
* shell 2: `sudo build/user_tools/begin_log`
* shell 2: `sudo touch /mnt/snapshot/test_dir/test_file`
* shell 2: `sudo chmod 0777 /mnt/snapshot/test_dir/test_file`
* shell 2: `echo "hello great big world out there" | sudo tee /mnt/snapshot/test_dir/test_file`
* shell 2: `sudo build/user_tools/end_log`
* shell 2: `sudo build/user_tools/begin_tests`

Note also that the placement of files to run has changed some. The new binary for the test harness is `build/harness`.

A later PR will fix the permissions issue cause all commands to need `sudo`. This will require fixing permissions on the socket itself.

### CODE REVIEW

**Comment by vijay03** [comment]
Good work! 

Why are the invocations for commands `begin log` and `end log` different? Both should be called in the same way, using a similar path. Are they at different locations? 

Could you add documentation for this into the README? Adding a separate page to the wiki also would be fine.

**Comment by ashmrtn** [comment]
That is just a typo in my comment apparently. They should all be in the same location. All of the functions to communicate with the CrashMonkey process use the same underlying code, they just seems different constants. I will fix that when I get a chance

File: code/harness/c_harness.cpp [hunk 1]
```diff
@@ -183,23 +259,48 @@ int main(int argc, char** argv) {
...
+            delete background_com;
+            test_harness.cleanup_harness();
+            return -1;
+          }
+          background_com->CloseClient();
+        }
+      } while (command != HARNESS_BEGIN_LOG);
+    } else {
+      // Run pre-test setup stuff.
+      cout << "Running pre-test setup\n";
```
Comment (vijay03) [inline code/harness/c_harness.cpp]: Make these visually distinct like:

=================
Running pre-test setup
=================

**Comment by ashmrtn** [comment]
Oh, haha sure thing! that should be easy since I kind of delineated that in the last commit.

**Pull request merged**